OpenRA.Game/Map/Map.cs

```diff
@@ -650,21 +650,24 @@ public void Save(IReadWritePackage toPackage)
 				foreach (var file in Package.Contents)
 					toPackage.Update(file, Package.GetStream(file).ReadAllBytes());
 
-			if (!LockPreview)
+			void UpdatePackage(string filename, byte[] data)
 			{
-				var previewData = SavePreview();
-				if (Package != toPackage || !Enumerable.SequenceEqual(previewData, Package.GetStream("map.png").ReadAllBytes()))
-					toPackage.Update("map.png", previewData);
+				if (Package != toPackage)
+					toPackage.Update(filename, data);
+				else
+				{
+					var stream = Package.GetStream(filename);
+					if (stream == null || !Enumerable.SequenceEqual(data, stream.ReadAllBytes()))
+						toPackage.Update(filename, data);
```

Comment:
Might as well compare two `Spans`? (Or even just always update?)

https://stackoverflow.com/questions/43289/comparing-two-byte-arrays-in-net

Reply:
We don't want to always update as to avoid changing the file edit date, see #20630. When it comes to spans wouldn't this be too much data?

Reply:
`Enumerable.SequenceEqual` will drop to comparing spans internally if it is possible. So I'd keep this as-is for readability.